Rog/src/main/java/springbootMigracion/java/com/example/demo/controller/InstrumentoFinancieroController.java

```diff
@@ -0,0 +1,83 @@
+package springbootMigracion.java.com.example.demo.controller;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import springbootMigracion.java.com.example.demo.dto.InstrumentoDTO;
+import springbootMigracion.java.com.example.demo.model.InstrumentoFinanciero;
+import springbootMigracion.java.com.example.demo.service.IInstrumentoFinancieroService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/instrumentos")
+@Slf4j
+public class InstrumentoFinancieroController implements IInstrumentoFinancieroApi {
+
+    @Autowired
+    private IInstrumentoFinancieroService instrumentoFinancieroService;
+
+    @PostMapping("/registrar")
+    public ResponseEntity<?> registrarInstrumento(@RequestBody InstrumentoDTO instrumentoDTO) throws Exception {
+        log.info("Solicitud para agregar instrumento: {}", instrumentoDTO);
+        InstrumentoFinanciero instrumento = instrumentoFinancieroService.registrarInstrumento(instrumentoDTO);
+        log.info("Instrumento agregado exitosamente: {}", instrumento);
+        return ResponseEntity.status(HttpStatus.CREATED).body(instrumento);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<InstrumentoFinanciero>> listarInstrumentos() {
+        List<InstrumentoFinanciero> instrumentos = instrumentoFinancieroService.listarTodosLosInstrumentos();
+        log.info("Instrumentos listados: {}", instrumentos.size());
+        return ResponseEntity.ok(instrumentos);
+    }
+
+    @GetMapping("/buscar/{nombre}")
+    public ResponseEntity<List<InstrumentoFinanciero>> listarInstrumentosPorNombre(@PathVariable String nombre) {
+        List<InstrumentoFinanciero> instrumentosPorNombre = instrumentoFinancieroService.listarInstrumentosPorNombre(nombre);
+        log.info("Instrumentos encontrados por nombre: {}", instrumentosPorNombre.size());
+        return ResponseEntity.ok(instrumentosPorNombre);
+    }
```

Comment:
El filtro/busqueda deberia de ser un QueryParam y no un path param, igualmnete esto deberia de estar unificado en el Get de Listar todos teniendo las 2 posibilidades, si hay o no RequestParams incluidos en la request dado que son opcionales.

```suggestion
    @GetMapping("")
    public ResponseEntity<List<InstrumentoFinanciero>> listarInstrumentosPorNombre(@RequestParam String nombre) {
        List<InstrumentoFinanciero> instrumentosPorNombre = instrumentoFinancieroService.listarInstrumentosPorNombre(nombre);
        log.info("Instrumentos encontrados por nombre: {}", instrumentosPorNombre.size());
        return ResponseEntity.ok(instrumentosPorNombre);
    }
```